### ISSUE TITLE
Mish inplace question

### ISSUE THREAD

**Issue by glenn-jocher** [rwightman/gen-efficientnet-pytorch#16]
Hi, thanks for the great repo! What's the reason that the `x.mul_(inner)` line is commented out? Does the commented line return different results? Thanks!

https://github.com/rwightman/gen-efficientnet-pytorch/blob/d2eadab6eb4427014c3b80eeaff1ff82d2d1f3f4/geffnet/activations/activations.py#L20-L25

**Comment by rwightman** [comment]
It's dead code now, I should get rid of it.. when I first implemented mish I though I'd be smart and make it a little more efficient with the inplace variation... however it triggered an inplace gradient error (during training)

**Comment by glenn-jocher** [comment]
Oh, that's strange. I trained with this same implementation without any errors, but got slightly worse results than Swish, wasn't sure if the inplace operation had anything to do with it.
https://github.com/ultralytics/yolov3/blob/429d44282cec6f07640057ff7782f5b751b02e4a/models.py#L124

```
class Mish(nn.Module):  # https://github.com/digantamisra98/Mish
    def __init__(self):
        super().__init__()

    def forward(self, x):
        return x.mul_(F.softplus(x).tanh())
```

**Comment by rwightman** [comment]
@glenn-jocher Maybe enabling anomaly_detection would let you know if there was a silent failure wrt to the gradient calc?

**Comment by glenn-jocher** [comment]
@rwightman that's an interesting idea. How is it enabled exactly?

**Comment by rwightman** [comment]
I believe you put your training loop, or call to it under a detect_anomaly context manager... been a while since I've used it, has a performance impact but this sort of error should be detected quickly.

```
with autograd.detect_anomaly():
    training_loop()
```
